src/Microsoft.Azure.WebJobs.StorageProvider.Common/StorageClientProvider.cs

```diff
@@ -0,0 +1,222 @@
+        public virtual bool TryGet(string name, out TClient client)
+        {
+            try
+            {
+                client = Get(name);
+                return true;
+            }
+            catch (Exception ex)
+            {
+                client = default(TClient);
+                _logger.LogError(ex, "Could not create Storage Client");
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Gets the storage client specified by <paramref name="connectionString"/>
+        /// </summary>
+        /// <param name="connectionString">Connection string used to authenticate requests</param>
+        /// <param name="client">Client that was created if the connection was valid; otherwise, the default value for the type of <paramref name="client"/>.</param>
+        /// <returns>true if the client could be created; false otherwise</returns>
+        public virtual bool TryGetFromConnectionString(string connectionString, out TClient client)
+        {
+            try
+            {
+                client = CreateClient(connectionString);
+                return true;
+            }
+            catch (Exception ex)
+            {
+
+                client = default(TClient);
+                _logger.LogError(ex, "Could not create Storage Client from a connection string.");
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Gets the storage client specified by <paramref name="name"/>
+        /// </summary>
+        /// <param name="name">Name of the connection to use</param>
+        /// <param name="resolver">A resolver to interpret the provided connection <paramref name="name"/>.</param>
+        /// <returns>Client that was created.</returns>
+        public virtual TClient Get(string name, INameResolver resolver)
+        {
+            var resolvedName = resolver.ResolveWholeString(name);
+            return this.Get(resolvedName);
+        }
+
+        /// <summary>
+        /// Gets the storage client specified by <paramref name="name"/>
+        /// </summary>
+        /// <param name="name">Name of the connection to use</param>
+        /// <returns>Client that was created.</returns>
+        public virtual TClient Get(string name)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                name = ConnectionStringNames.Storage; // default
+            }
+
+            // $$$ Where does validation happen?
+            IConfigurationSection connectionSection = _configuration.GetWebJobsConnectionStringSection(name);
+            if (!connectionSection.Exists())
+            {
+                // Not found
+                throw new InvalidOperationException($"Storage account connection string '{IConfigurationExtensions.GetPrefixedConnectionStringName(name)}' does not exist. Make sure that it is a defined App Setting.");
+            }
+
+            _logForwarder.Start();
```

Comment:
Will this type handle multiple calls to `Start` gracefully?

Reply:
It should, but moving this to the constructor to lessen the calls to it.